tests/python/contrib/test_ethosu/cascader/test_memory_reduction.py

```diff
@@ -0,0 +1,211 @@
+    pool_size = 2000000
+    workspace_size_cascader_disabled = _get_ethosu_workspace_size(
+        mod, params, accel_type, pool_size, enable_cascader=False
+    )
+
+    # Run the same graph with the cascader, giving it less memory to persuade cascder to cascade
+    pool_size = 600000
+    workspace_size_cascader_enabled = _get_ethosu_workspace_size(
+        mod, params, accel_type, pool_size, enable_cascader=True
+    )
+
+    assert workspace_size_cascader_enabled < workspace_size_cascader_disabled
+
+
+# TODO(ekalda): Fix a bug in the block config selection that selects block config that is too large
+# for the smaller accelerators
+@pytest.mark.parametrize(
+    "accel_type",
+    (
+        "ethos-u55-256",
+        "ethos-u55-128",
+        pytest.param("ethos-u55-64", marks=pytest.mark.xfail),
+        pytest.param("ethos-u55-32", marks=pytest.mark.xfail),
+    ),
+)
+def test_depthwise2d_conv2d_pooling(accel_type):
+    np.random.seed(2)
+    ifm_shape = (1, 80, 75, 3)
+
+    @tf.function
+    def tf_graph(x):
+        # This graph will execute as one cascade
+        ofm_channels = 7
+        conv2d = tf.nn.conv2d(
+            x,
+            filters=tf.constant(
+                np.random.uniform(size=[3, 2, ifm_shape[3], ofm_channels]),  # HWIO
+                dtype=tf.float32,
+            ),
+            strides=(1, 1),
+            padding="VALID",
+            dilations=(1, 1),
+        )
+        depthwise2d = tf.nn.depthwise_conv2d(
+            conv2d,
+            tf.constant(np.random.uniform(size=(3, 3, ofm_channels, 1)), dtype=tf.float32),  # HWC1
+            strides=(1, 1, 1, 1),
+            padding="VALID",
+            dilations=(1, 1),
+        )
+        relu = tf.nn.relu(depthwise2d)
+        conv2d = tf.nn.conv2d(
+            relu,
+            filters=tf.constant(
+                np.random.uniform(size=[3, 2, ofm_channels, 2]),  # HWIO
+                dtype=tf.float32,
+            ),
+            strides=(1, 1),
+            padding="SAME",
+            dilations=(1, 1),
+        )
+        max_pool = tf.nn.max_pool(conv2d, (3, 3), (1, 1), "SAME")
+
+        return max_pool
+
+    _, tflite_graph = infra.get_tflite_graph(tf_graph, [ifm_shape])
+    tflite_model = tflite.Model.Model.GetRootAsModel(tflite_graph, 0)
+
+    relay_module, params = relay.frontend.from_tflite(tflite_model)
+    mod = partition_for_ethosu(relay_module, params)
+
+    # Run the graph without the cascader, with lots of memory
+    pool_size = 10 ** 6
+    workspace_size_cascader_disabled = _get_ethosu_workspace_size(
+        mod, params, accel_type, pool_size, enable_cascader=False
+    )
+
+    # Run the same graph with the cascader, giving it less memory to persuade cascder to cascade
+    pool_size = 40000
+    workspace_size_cascader_enabled = _get_ethosu_workspace_size(
+        mod, params, accel_type, pool_size, enable_cascader=True
+    )
+
+    assert workspace_size_cascader_enabled < workspace_size_cascader_disabled
```

Comment:
out of curiosity, should we not check for exact values ?

Reply:
Yes good point, I changed it to check for the exact values